python/packages/autogen-agentchat/src/autogen_agentchat/message_store/_memory_message_store.py

```diff
@@ -0,0 +1,69 @@
+import asyncio
+import time
+from collections import deque
+from typing import Optional, Sequence
+
+from pydantic import BaseModel, Field
+
+from ..messages import BaseAgentEvent, BaseChatMessage, TextMessage
+from ._message_store import MessageStore
+
+
+class MemoryMessage(BaseModel):
+    """A message stored in memory.
+
+    Args:
+        message (BaseAgentEvent | BaseChatMessage): The message to store.
+        timestamp (int): The timestamp of the message in seconds since epoch.
+    """
+
+    message: BaseAgentEvent | BaseChatMessage | TextMessage
+    ts: int = Field(default_factory=lambda: int(time.time()))
+
+
+class MemoryMessageStore(MessageStore):
```

Comment:
We need to subclass Component and implement the methods. See example in other classes like model context classes

Reply:
Thanks for kind feedback, ekzhu. Let me update what you said tonight.

Reply:
fixed via https://github.com/microsoft/autogen/pull/6350/commits/49167ae7280cf5f099260ec7caa8f482fad793a2